cfpb/cfgov-refresh
# Add complaint maintenance banner and upgrade Django-Flags

This PR adds a feature flag, a template, and a template tag for a complaint intake maintenance banner to sit on `/complaint` URLs. This is similar to #4745 and the service degradation banner, except for planned maintenance windows. 

The maintenance banner and its corresponding feature flag, `COMPLAINT_INTAKE_MAINTENANCE`, are intended to be used with Django-Flags "after date" and "before date" conditions to define the time window in which maintenance will take place. The banner uses the "before date" condition to inform users when the maintenance window will close and the complaint intake forms will be available again.

In doing this I've also simplified the implementation of the `COMPLAINT_INTAKE_TECHNICAL_ISSUES` flag to use required conditions in Django-Flags (https://github.com/cfpb/django-flags/pull/24) to require a particular matching path instead of requiring a separate setting of paths to match.

## Testing

1. Install the Django-Flags required-conditions branch in the Docker python container:
  
    ```shell
    ./shell.sh
    pip install git+https://github.com/cfpb/django-flags.git@required-conditions
   ```

2. Visit the flags admin at http://localhost:8000/admin/flags/

3. Add the following conditions:

   A required condition that will enable the flag after 2019/02/01 at midnight:

   - Flag: `COMPLAINT_INTAKE_MAINTENANCE`
   - Condition name: after date
   - Expected value: 2019-02-01T05:00Z
   - Required: Check the box

   A required condition that will disable the flag after 2019/03/01 at midnight:

   - Flag: `COMPLAINT_INTAKE_MAINTENANCE`
   - Condition name: before date
   - Expected value: 2019-03-01T05:00Z
   - Required: Check the box

   And finally a non-required condition that enable the flag if the required conditions are met:

   - Flag: `COMPLAINT_INTAKE_MAINTENANCE`
   - Condition name: boolean
   - Expected value: True
   - Required: DO NOT Check the box

4. Visit http://localhost:8000/complaint/ and see the banner:

![image](https://user-images.githubusercontent.com/10562538/52134207-0c0d2b80-2611-11e9-8c5a-263e8984f1ab.png)

## Notes:

This PR includes a non-required boolean condition set to `False` for both the complaint banners. Because the paths are now required feature flag conditions if it doesn't include a condition that disables it *with* the path, the banners will always appear. Clearly this is not desirable. Because the boolean condition is not required, when we add conditions that *do* enable to flag, they will be met and the flag will be enabled. The boolean condition merely achieves a default "off" state.

## Checklist
